src/jace/translated_jaxpr_sdfg.py

```diff
@@ -0,0 +1,229 @@
+    out_names: tuple[str, ...]
+
+    def __init__(
+        self,
+        csdfg: compiled_sdfg.CompiledSDFG,
+        inp_names: tuple[str, ...],
+        out_names: tuple[str, ...],
+    ) -> None:
+        self.csdfg = csdfg
+        self.sdfg = self.csdfg.sdfg
+        self.inp_names = inp_names
+        self.out_names = out_names
+
+    def __call__(
+        self,
+        flat_call_args: Sequence[Any],
+    ) -> list[np.ndarray] | None:
+        """
+        Run the compiled SDFG using the flattened input.
+
+        The function will not perform flattening of its input nor unflattening of
+        the output.
+
+        Args:
+            csdfg: The compiled SDFG to call.
+            flat_call_args: Flattened input arguments.
+        """
+        if len(self.inp_names) != len(flat_call_args):
+            # Either error or static arguments are not removed.
+            raise RuntimeError("Wrong number of arguments.")
+
+        sdfg_call_args: dict[str, Any] = {}
+        for in_name, in_val in zip(self.inp_names, flat_call_args):
+            # TODO(phimuell): Implement a stride matching process.
+            if util.is_jax_array(in_val):
+                if not util.is_fully_addressable(in_val):
+                    raise ValueError(f"Passed a not fully addressable Jax array as '{in_name}'")
+                in_val = in_val.__array__()  # noqa: PLW2901  # Jax arrays do not expose the __array_interface__.
+            sdfg_call_args[in_name] = in_val
+
+        arrays = self.sdfg.arrays
+        for out_name, sdfg_array in ((out_name, arrays[out_name]) for out_name in self.out_names):
+            if out_name in sdfg_call_args:
+                if util.is_jax_array(sdfg_call_args[out_name]):
+                    raise ValueError("Passed an immutable Jax array as output.")
+            else:
+                sdfg_call_args[out_name] = dace_data.make_array_from_descriptor(sdfg_array)
+
+        assert len(sdfg_call_args) == len(self.csdfg.argnames), (
+            "Failed to construct the call arguments,"
+            f" expected {len(self.csdfg.argnames)} but got {len(flat_call_args)}."
+            f"\nExpected: {self.csdfg.argnames}\nGot: {list(sdfg_call_args.keys())}"
+        )
+
+        # Calling the SDFG
+        with dace.config.temporary_config():
+            dace.Config.set("compiler", "allow_view_arguments", value=True)
+            self.csdfg(**sdfg_call_args)
+
+        if self.out_names:
+            return [sdfg_call_args[out_name] for out_name in self.out_names]
+        return None
+
+
+def compile_jaxpr_sdfg(tsdfg: TranslatedJaxprSDFG) -> CompiledJaxprSDFG:
+    """Compile `tsdfg` and return a `CompiledJaxprSDFG` object with the result."""
+    if any(  # We do not support the DaCe return mechanism
+        array_name.startswith("__return")
+        for array_name in tsdfg.sdfg.arrays.keys()  # noqa: SIM118  # We can not use `in` because we are not interested in `my_mangled_variable__return_zulu`!
```

Comment:
But what would be a better message, I have no idea.

Reply:
I was talking about the comment for the `noqa`. Our coding conventions say that `noqa` comments should be formatted as `# noqa: XXX0123 [long_ruff_name]  # this is ok because ...` and the current code doesn't respect this format nor explains why the linter's message should be ignored.

It seems to me that you didn't check the meaning of the error code (https://docs.astral.sh/ruff/rules/in-dict-keys/) and I don't see why it should be ignored, since it's saying that the `in` operator for dicts only checks if something appears in the keys of the dict, and thus the `.keys()` is not needed.